src/main/scala/mesosphere/marathon/api/akkahttp/v2/TasksController.scala

```diff
@@ -45,15 +52,32 @@ class TasksController(
     asLeader(electionService) {
       authenticated.apply { implicit identity =>
         get {
-          pathEndOrSingleSlash {
-            listTasks()
+          (accepts(MediaTypes.`text/plain`) & pathEndOrSingleSlash) {
+            listTasksTxt()
+          } ~
+          (accepts(MediaTypes.`application/json`) & pathEndOrSingleSlash) {
+            listTasksJson()
+          } ~
+          acceptsAnything {
+            listTasksJson() // when no accept header present, json is the default choice
           }
         }
       }
     }
   }
 
-  private def listTasks()(implicit identity: Identity): Route = {
+  @SuppressWarnings(Array("all")) // async/await
+  private def listTasksTxt()(implicit identity: Identity): Route = {
```

Comment:
How come this method differs so much from `listTasksJson`?

Reply:
I would say that much less data is actually serialized to the text output than to json ... so even the processing and preparation of the data is a little bit different...

Reply:
Ok, thanks for clearing this up.